src/nvim/strings.c

```diff
@@ -979,7 +979,6 @@ int vim_vsnprintf(char *str, size_t str_m, const char *fmt, va_list ap,
           const void *ptr_arg = NULL;
 
           if (fmt_spec == 'p') {
-            length_modifier = '\0';
```

Comment:
I agree that it should be safe to remove. This code is within a while-loop, but `length_modifier` is declared and initialized within the loop, so only the code below this statement needs to be checked.